lib/xhr/XHRShim.js

```diff
@@ -0,0 +1,142 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, indent: 4, forin: true, maxerr: 50, regexp: true */
+/* global DOMParser */
+(function(global) {
+    "use strict";
+
+    // This script requires access to the transport to send
+    // file requests to the editor as commands
+    var transport = global._Brackets_LiveDev_Transport;
+
+    var XMLHttpRequest = global.XMLHttpRequest;
+    if(!XMLHttpRequest) {
+        return;
+    }
+
+    function sendMessage(msg) {
+        if(!transport || !transport.send) {
+            console.error("[Brackets LiveDev] No transport set");
+            return;
+        }
+
+        transport.send(JSON.stringify(msg));
+    }
+
+    function XMLHttpRequestLiveDev() {
+        var self = new XMLHttpRequest();
+        var requestUrl;
+        var abortCalled = false;
+
+        var open = self.open;
+        // The async parameter for XHRs to the local file system
+        // is ignored as we cannot support synchronous requests
+        // and async is implicit
+        self.open = function(method, url, async, user, password) {
+            if(url.indexOf("//") === -1) {
```

Comment:
Let's use https://github.com/humphd/brackets/blob/bramble/src/filesystem/impls/filer/lib/content.js#L104 here instead of rewriting this code.